contracts/mainnet/connectors/euler/euler-import/main.sol

```diff
@@ -0,0 +1,157 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+import "./helpers.sol";
+import "./interface.sol";
+import "./events.sol";
+
+contract EulerImport is EulerHelpers {
+	/**
+	 * @dev Import Euler position .
+	 * @notice Import EOA's Euler position to DSA's Euler position
+	 * @param userAccount EOA address
+	 * @param sourceId Sub-account id of "EOA" from which the funds will be transferred
+	 * @param targetId Sub-account id of "DSA" to which the funds will be transferred
+	 * @param inputData The struct containing all the neccessary input data
+	 */
+	function importEuler(
+		address userAccount,
+		uint256 sourceId,
+		uint256 targetId,
+		ImportInputData memory inputData
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		(_eventName, _eventParam) = _importEuler(
+			userAccount,
+			sourceId,
+			targetId,
+			inputData
+		);
+	}
+
+	/**
+	 * @dev Import Euler position .
+	 * @notice Import EOA's Euler position to DSA's Euler position
+	 * @param userAccount EOA address
+	 * @param sourceId Sub-account id of "EOA" from which the funds will be transferred
+	 * @param targetId Sub-account id of "DSA" to which the funds will be transferred
+	 * @param inputData The struct containing all the neccessary input data
+	 */
+	function _importEuler(
+		address userAccount,
+		uint256 sourceId,
+		uint256 targetId,
+		ImportInputData memory inputData
+	) 
+        internal 
+        returns (string memory _eventName, bytes memory _eventParam) 
+    {
+
```

Comment:
We also need check that `sourceId` and `targetId` should be less than 256.

Reply:
Added